src/readblock.c

```diff
@@ -0,0 +1,299 @@
+/****************************************************************************************
+
+    FFIO - Flexible format I/O
+
+    FFIO provides functionality for reading proj style instruction strings written
+    in a less strict format than usual:
+
+    *  Whitespace is generally allowed everywhere
+    *  Comments can be written inline, '#' style
+    *  ... or as free format blocks
+
+    The overall mission of FFIO is to facilitate communications of geodetic
+    parameters and test material in a format that is highly human readable,
+    and provides ample room for comment, documentation, and test material.
+
+    See the PROJ ".gie" test suites for examples of supported formatting.
+
+****************************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+
+#include <string.h>
+#include <ctype.h>
+
+#include <math.h>
+#include <errno.h>
+
+
+
+static const char *gie_tags[] = {
+    "<gie>", "operation", "accept", "expect", "roundtrip", "banner", "verbose",
+    "direction", "tolerance", "builtins", "echo", "</gie>"
+};
+
+static const size_t n_gie_tags = sizeof gie_tags / sizeof gie_tags[0];
```

Comment:
Is there no array size macro for Proj?

Reply:
No - and I think this is the only place where one would be needed